theme/additional.js

```diff
@@ -36,3 +36,33 @@
     }
   });
 })();
+
+function setCodeToPlayground() {
+  var codes = JSON.parse(localStorage.getItem(window.location.href));
+  if (codes) {
+    var i = 0;
+    Array.from(document.querySelectorAll(".playground")).forEach(function (
+      pre_block
+    ) {
+      let code_block = pre_block.querySelector("code");
+      let editor = window.ace.edit(code_block);
+      editor.setValue(codes[i]);
+      editor.clearSelection();
+      i += 1;
+    });
+  }
+}
+setCodeToPlayground();
```

Comment:
minor point: this is a bit hard to spot here. Please move the whole thing to an IIFE, like the red-box functionality, and put the immediately-evaluated statements at the bottom:
```js
(function savePlaygrounds() {
  function setCodeToPlayground() ..
  function getCodeFromPlaground() ..

  setCodeToPlayground();
  setInterval(getCodeFromPlayground, 900);
})()
```